src/BlockArea.cpp

```diff
@@ -673,36 +673,22 @@ void cBlockArea::Crop(int a_AddMinX, int a_SubMaxX, int a_AddMinY, int a_SubMaxY
 	}
 	if (HasBlockEntities())
 	{
-		auto maxX = m_Size.x - a_SubMaxX;
-		auto maxY = m_Size.y - a_SubMaxY;
-		auto maxZ = m_Size.z - a_SubMaxZ;
+		const Vector3i AddMin{ a_AddMinX, a_AddMinY, a_AddMinZ };
+		const cCuboid CropBox{ AddMin, m_Size - Vector3i{a_SubMaxX, a_SubMaxY, a_SubMaxZ} };
 
 		// Move and crop block Entities:
 		cBlockEntities oldBE;
 		std::swap(oldBE, *m_BlockEntities);
```

Comment:
I admit I don't know how BlockAreas manage block entities so I can't make any suggestions. Perhaps @madmaxoft can shed some light; why are block entities being deleted here? I thought they were World-owned?

Reply:
BEs in a `cBlockArea` are owned by the `cBlockArea`. They are clones of the ones in the `cWorld` from which they were read (if at all), and when writing to a `cWorld`, they are again cloned.

Reply:
Thanks. Is it a possibility to use the world ones directly and not clones? At a minimum there shouldn't be threading issues.

Reply:
No, because you don't expect that changing a BE in a BlockArea to change a BE in the World. There's no COW.